homepage/models.py

```diff
@@ -125,10 +184,64 @@ def get_elective_courses(courses):
         # gets all courses that are electives
         return courses.filter(mandatory=False)
 
-    # --- returns if course has Prerequisites
-    def has_preqs(self):
-        return Prerequisites.does_course_have_prerequisites(self)
+    @staticmethod
+    def get_courses_with_preqs(courses):
+        # gets all courses that have prerequisites
+        have_preqs = Course.get_courses_with_preqs_ids(courses)
+        return courses.filter(course_id__in=have_preqs)
+
+    @staticmethod
+    def get_courses_with_preqs_ids(courses):
+        have_preqs = []
+        for course in courses:
+            if course.has_preqs():
+                have_preqs.append(course.course_id)
+        return have_preqs

```

Comment:
you can use list comprehension
```suggestion
        return [ course.course_id for course in courses if course.has_preqs() ]
```

Reply:
The courses page is quite hinged on the all_courses being a QuerySet, so I would like to keep it as it is.